### ISSUE TITLE
Animations

### ISSUE THREAD

**Issue by OlaHulleberg** [sivanmah/sunshine-amigos#9]
Assign yourself to this issue if you want to work on it.
Make sure you work on a branch seperate from the main branch, and open a pull request when ready to merge.

**Comment by OlaHulleberg** [comment]
We should start working on the different animations, so we can integrate them later.
If you have any suggestions or thoughts on the animations described by the story board, post them here.

**Comment by elizaveta-sergejevna** [comment]
What are we using for the animations? Pure CSS or...?

**Comment by OlaHulleberg** [comment]
Yes, keyframes and so on that can be applied by appending classes to the different elements. F.i .verse2-fade-in or .verse2-winter-colors and so on

**Comment by elizaveta-sergejevna** [comment]
Okay, and how about the tree and the different versions of it? When transitioning into a human silhouette, for instance, do we need an SVG-image of the silhouette and link it to the index.html via `<img>`? And then working with the animations between the different images aka the different stages of the tree? (I might be asking in a very difficult manner; I simply want to know the basic technicalities of how to animate the darn tree :stuck_out_tongue_closed_eyes:)

**Comment by OlaHulleberg** [comment]
I mean, there is no set way to do it. You kind of figure out as you go (At least that's what I do, by trial & failure). I have made a tree-container that has the ability to have several svgs overlap each other. All you (in theory) should have to do is to change the z-index of your new SVG to some value that doesn't correspond to another SVG, and it should work with minor tweaks. Illustrator should have the possibility of opening SVG files. 

Here's an example of how you can use it to animate SVG files: [Creating a vector animation for the web in a post-Flash world](https://thislittleduck.com/blog/svg-animation-using-snap-svg/).
The animation is at the bottom of the screen, and is activated by the code she supplied in Adobe Animate, which can easily just be converted to something like this instead:

```
var messageClip = this.messageClip;

function playMyCoolAnimation() {
    messageClip.gotoAndPlay(2); // Or whatever clip you need to go to
}
```

But as I said, there is no set way to do anything here. If you have another completely different solution that animates the tree, then great, we can use that as well! :)
To find this I just Googled "Adobe Illustrator animate svg", and it was the first result to pop up.

TL;DR: Yes, using `<img>`. If you open an SVG, you'll find the same lines of code as you put in the HTML earlier. It's just a seperate file to make things tidier.

You can even find this at the bottom of our SVG files:
`<!-- Generator: Adobe Illustrator 25.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->`

**Comment by OlaHulleberg** [comment]
If you need to replace one of the SVGs with another in an animation, I recommend that both of the SVGs have the same overall image size. Although it shouldn't strictly be needed.

**Comment by OlaHulleberg** [comment]
You can also do all of these animations by hand, using the CSS classes as described in my first reply.

**Comment by elizaveta-sergejevna** [comment]
I understand that there are a million ways of creating animations; I just thought it would be best if everyone agreed upon _one_ method so that our work is consistent. Surely the animations will have to be made by several people and not by one individual only. 

I guess I can try to experiment a bit and see where things go :metal:

**Comment by elizaveta-sergejevna** [comment]
Alrighty then, even though I didn't want to, I've taken upon myself to finish the rest of the animations. I'm sure they could've been better and more advanced, but at least we have something :cowboy_hat_face: Please take a closer look at my comments in _mankind.scss, _frame9.scss and _frame10.scss. If you have any feedback, I need to know about it ASAP (or you may of course try to fix any issues yourself :sparkling_heart:). @OlaHulleberg @sivanmah @williambaardvik

**Comment by OlaHulleberg** [comment]
You've pushed these to the master branch, so the changes are sort of already approved x)
I really think these last frames are _the_ **B.O.M.B** (_that's good_)!
I might try to play a little bit with the positioning of the human(s), and maybe enable the "disco" mode for the ground portion too.

Great job!

**Comment by elizaveta-sergejevna** [comment]
**OOPSIE!** I thought that since we were starting merging branches and whatnot, that that was okay :grimacing: But I mean, you could just revert my commits or make changes directly in the branch itself, no? 

I think we might be close to the finish line, so I hope that whatever I did, we can just work around it until the end :sweat_smile:

Thank you for your kind words; I'm really proud of the disco mode :sunglasses: (which is what color transitions shall be called from now on).

**Comment by OlaHulleberg** [comment]
I've made some suggestive changes in the animation branch, would you take a look? @elizaveta-sergejevna

**Comment by OlaHulleberg** [comment]
I've also fixed the infinite pumping issue @elizaveta-sergejevna 

In short: `animation-iteration-count` allows for multiple parameters/arguments, where each corresponds (in order) to the `animation-name` listed above (or below)

**Comment by elizaveta-sergejevna** [comment]
I saw your changes and raised with one small change of my own. Check it out and pay attention to the scrolling :call_me_hand:
